### ISSUE TITLE
macos14.4 switched from perl5.30(.3) to perl5.34(.1)

### ISSUE THREAD

**Issue by nieder** [fink/fink#276]
From comment https://github.com/fink/fink-distributions/commit/54d4220054f6e45299367556d99860c40b897d01#r150409884

Then, following links from https://opensource.apple.com/releases/

macOS 14.0-14.3 has apple-perl-155, which has perl5.30 as the default
https://github.com/apple-oss-distributions/perl/blob/perl-155/Platforms/MacOSX/Makefile.inc
This is what we have been using.

macOS 14.4 has perl-157.100.1, which has perl5.34 as the default
https://github.com/apple-oss-distributions/perl/blob/perl-157.100.1/Platforms/MacOSX/Makefile.inc
This is not what our code expects.

So we'll need a new distribution called 14.4 with perl5.34 (similar to what happened with the 11.2 -> 11.3 transition).
I'll try to get to this tonight.

**Comment by dhomeier** [comment]
Does this mean on 14.0-14.3 perl5.34 may exist, but not be the default? Currently 5608343 assumes that whenever `/usr/bin/perl5.34` exists on 14+ it also is `systemperl`; that may need a more precise version check then.

**Comment by nieder** [comment]
yes, /usr/bin/perl5.34 will probably be present. It's not the first time that multiple perls have existed at the same time. The one we consider the default that defines the Distribution is whatever /usr/bin/perl points to. We've had this before with 10.14.5 and 11.3:

https://github.com/fink/fink/blob/568033f0c6ad59633fe1ce3959704cd07a677748/perlmod/Fink/PkgVersion.pm#L5830-L5834

**Comment by dhomeier** [comment]
So if `/usr/bin/perl5.34` exists but `/usr/bin/perl` is 5.30, `$perlversion` will still be 5.30.3 – then it would actually still point to the correct default?

**Comment by nieder** [comment]
That's the expectation, and I have a 12.7 system that has /usr/bin/pel5.30, /usr/bin/perl -> perl5.30 as well /usr/bin/perl5.18, and it always uses perl5.30.
